packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -98,9 +99,15 @@ class PlaceholderDimensions {
   ///  * [ui.PlaceholderAlignment.middle]
   final TextBaseline? baseline;
 
+  /// The range of text that this placeholder replaces.
+  ///
+  /// This value is only relevant when used in TextFields/Editable widgets
+  /// as replacements for strings of regular text.
+  final TextRange? range;
```

Comment:
Should this just be TextRange.empty in the cases where it is not relevant?

Reply:
(That may remove the need for some of the `!` below)

Reply:
done!